overlord/snapstate/policy/canremove_test.go

```diff
@@ -154,12 +154,7 @@ func (s *canRemoveSuite) TestOSNoSnapdNotOK(c *check.C) {
 		Current:  snap.R(1),
 		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{{Revision: snap.R(1), RealName: "core"}}),
 	}
-	c.Check(policy.NewOSPolicy("").CanRemove(s.st, snapst, snap.R(1), classicDev), check.Equals, policy.ErrSnapdNotInstalled)
-
-	snapst = &snapstate.SnapState{
-		Current:  snap.R(1),
-		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{{Revision: snap.R(1), RealName: "core"}}),
-	}
+	// revision is unset as if we're fully removing core from the system
 	c.Check(policy.NewOSPolicy("").CanRemove(s.st, snapst, snap.Revision{}, classicDev), check.Equals, policy.ErrSnapdNotInstalled)
```

Comment:
do we need a test for the other case where we are removing one revision of many?

Reply:
There already was one for removing a specific revision from core systems. I added a case for classic systems.